### PR TITLE
fix ubsan test errors

### PR DESCRIPTION
**Pull request by Cyan4973**
allows recovering from pointer overflow.

### CODE REVIEW

**Review by terrelln — APPROVED** [review]
FYI this won't work with gcc versions older than `8.0.0`, but that shouldn't be a problem.

**Comment by terrelln** [comment]
I've fixed the LLVM fuzzers in https://github.com/facebook/zstd/pull/2503.

File: Makefile [hunk 1]
```diff
@@ -309,10 +309,10 @@ asan32: clean
 	$(MAKE) -C $(TESTDIR) test32 CC=clang MOREFLAGS="-g -fsanitize=address"
 
 uasan: clean
-	$(MAKE) test CC=clang MOREFLAGS="-g -fno-sanitize-recover=all -fsanitize-recover=signed-integer-overflow -fsanitize=address,undefined -Werror"
+	$(MAKE) test CC=clang MOREFLAGS="-g -fno-sanitize-recover=all -fsanitize-recover=signed-integer-overflow -fsanitize-recover=pointer-overflow -fsanitize=address,undefined -Werror"
```
Comment (terrelln) [inline Makefile]: Can we remove `-fsanitize-recover=signed-integer-overflow`?

Comment (Cyan4973) [reply Makefile]: Yes.
At least on my local platform, it works fine.

**Pull request merged**